SoftLayer/CLI/firewall/detail.py

```diff
@@ -12,31 +12,67 @@
 
 @click.command(cls=SoftLayer.CLI.command.SLCommand, )
 @click.argument('identifier')
+@click.option('--credentials', type=click.BOOL,
+              help="Display FortiGate username and FortiGate password to multi vlans.")
 @environment.pass_env
-def cli(env, identifier):
+def cli(env, identifier, credentials):
     """Detail firewall."""
 
     mgr = SoftLayer.FirewallManager(env.client)
 
     firewall_type, firewall_id = firewall.parse_id(identifier)
-    _firewall = mgr.get_instance(firewall_id)
 
-    table = formatting.KeyValueTable(['name', 'value'])
-    table.align['name'] = 'r'
-    table.align['value'] = 'l'
+    if firewall_type == 'vs' or firewall_type == 'server' or firewall_type == 'vlan' or firewall_type == 'multiVlan':
 
-    table.add_row(['id', _firewall.get('id')])
-    table.add_row(['primaryIpAddress', _firewall.get('primaryIpAddress')])
-    table.add_row(['datacenter', utils.lookup(_firewall, 'datacenter', 'longName')])
-    table.add_row(['networkVlan', utils.lookup(_firewall, 'networkVlan', 'name')])
-    table.add_row(['networkVlaniD', utils.lookup(_firewall, 'networkVlan', 'id')])
+        if firewall_type == 'vlan':
+            _firewall = mgr.get_instance(firewall_id)
+
+            table = formatting.KeyValueTable(['name', 'value'])
+            table.align['name'] = 'r'
+            table.align['value'] = 'l'
+
+            table.add_row(['id', _firewall.get('id')])
+            table.add_row(['primaryIpAddress', _firewall.get('primaryIpAddress')])
+            table.add_row(['datacenter', utils.lookup(_firewall, 'datacenter', 'longName')])
+            table.add_row(['networkVlan', utils.lookup(_firewall, 'networkVlan', 'name')])
+            table.add_row(['networkVlaniD', utils.lookup(_firewall, 'networkVlan', 'id')])
+
+            rules = mgr.get_dedicated_fwl_rules(firewall_id)
+            table.add_row(['rules', get_rules_table(rules)])
+
+        if firewall_type == 'multiVlan':
+            _firewall = mgr.get_instance(firewall_id)
+
+            table = formatting.KeyValueTable(['name', 'value'])
+            table.align['name'] = 'r'
+            table.align['value'] = 'l'
+
```

Comment:
uppercase or lowercase, please use the same rule on the names of table

Reply:
the names of of table were changed to lowercase